sdc/datatypes/common_functions.py

```diff
@@ -641,3 +644,59 @@ def _almost_equal_impl(x, y):
         return abs(x - y) <= numpy.finfo(common_dtype).eps
 
     return _almost_equal_impl
+
+
+def sdc_reindex_series(arr, index, name, by_index):
+    pass
+
+
+@sdc_overload(sdc_reindex_series, jit_options={'parallel': True})
+def sdc_reindex_series_overload(arr, index, name, by_index):
+    """ Reindexes series data by new index following the logic of pandas.core.indexing.check_bool_indexer """
+
+    data_dtype, index_dtype = arr.dtype, index.dtype
+    data_is_str_arr = isinstance(arr.dtype, types.UnicodeType)
+
+    def sdc_reindex_series_impl(arr, index, name, by_index):
+        res_index = by_index.copy()
+        if data_is_str_arr == True:  # noqa
+            res_data_as_list = [''] * len(by_index)
+            res_data_nan_mask = numpy.zeros(len(by_index), dtype=types.bool_)
+        else:
+            res_data = numpy.empty(len(by_index), dtype=data_dtype)
+
+        # build a dict of self.index values to their positions:
+        map_index_to_position = Dict.empty(
+            key_type=index_dtype,
+            value_type=types.int32
+        )
+
+        for i, value in enumerate(index):
+            if value in map_index_to_position:
+                raise ValueError("cannot reindex from a duplicate axis")
+            else:
+                map_index_to_position[value] = i
+
+        index_mismatch = 0
```

Comment:
Looks `index_mismatch` could be of a boolean type.

Reply:
@densmirn Not sure if it's really possible, because it's used inside prange and simply assigning  True value would be a race between threads. I prefer to leave it as is, so that it's reduced to a number of faults occurred during parallel iteration.